pkg/status/status.go

```diff
@@ -0,0 +1,187 @@
+func SetStatusConditions(ctx context.Context, client runtimeclient.StatusClient, logger logr.Logger, object interface{}, conditions *kedav1alpha1.Conditions) error {
+	transform := func(runtimeObj runtimeclient.Object, target interface{}) error {
+		conditions, ok := target.(*kedav1alpha1.Conditions)
+		if !ok {
+			return fmt.Errorf("transform target is not kedav1alpha1.Conditions type %v", target)
+		}
+		switch obj := runtimeObj.(type) {
+		case *kedav1alpha1.ScaledObject:
+			obj.Status.Conditions = *conditions
+		case *kedav1alpha1.ScaledJob:
+			obj.Status.Conditions = *conditions
+		default:
+		}
+		return nil
+	}
+	return TransformObject(ctx, client, logger, object, conditions, transform)
+}
+
+// UpdateScaledObjectStatus patches the given ScaledObject with the updated status passed to it or returns an error.
+func UpdateScaledObjectStatus(ctx context.Context, client runtimeclient.StatusClient, logger logr.Logger, scaledObject *kedav1alpha1.ScaledObject, status *kedav1alpha1.ScaledObjectStatus) error {
+	transform := func(runtimeObj runtimeclient.Object, target interface{}) error {
+		status, ok := target.(*kedav1alpha1.ScaledObjectStatus)
+		if !ok {
+			return fmt.Errorf("transform target is not kedav1alpha1.ScaledObjectStatus type %v", target)
+		}
+		switch obj := runtimeObj.(type) {
+		case *kedav1alpha1.ScaledObject:
+			obj.Status = *status
+		default:
+		}
+		return nil
+	}
+	return TransformObject(ctx, client, logger, scaledObject, status, transform)
+}
+
+func getTriggerAuth(ctx context.Context, client runtimeclient.Client, triggerAuthRef *kedav1alpha1.ScaledObjectAuthRef, namespace string) (runtimeclient.Object, *kedav1alpha1.TriggerAuthenticationStatus, error) {
+	if triggerAuthRef == nil {
+		return nil, nil, fmt.Errorf("triggerAuthRef is nil")
+	}
+
+	if triggerAuthRef.Kind == "" || triggerAuthRef.Kind == "TriggerAuthentication" {
+		triggerAuth := &kedav1alpha1.TriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, triggerAuth)
+		if err != nil {
+			return nil, nil, err
+		}
+		return triggerAuth, &triggerAuth.Status, nil
+	} else if triggerAuthRef.Kind == "ClusterTriggerAuthentication" {
+		clusterTriggerAuth := &kedav1alpha1.ClusterTriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, clusterTriggerAuth)
+		if err != nil {
+			return nil, nil, err
+		}
+		return clusterTriggerAuth, &clusterTriggerAuth.Status, nil
+	}
+	return nil, nil, fmt.Errorf("unknown trigger auth kind %s", triggerAuthRef.Kind)
+}
+
+func updateTriggerAuthenticationStatus(ctx context.Context, logger logr.Logger, client runtimeclient.Client, namespace string, triggerAuthRef *kedav1alpha1.ScaledObjectAuthRef, statusHandler func(*kedav1alpha1.TriggerAuthenticationStatus) *kedav1alpha1.TriggerAuthenticationStatus) error {
```

Comment:
could you please add a short doc comment here on what this function does and returns?